Fixes #35921 - Unify use of AccessLog

adamruzicka opened the pull request:
across http and https applications

adamruzicka commented:
Should we unify on on access log being turned off or on? Alternatively we could do it conditionally based on logging configuration

ekohl commented on lib/launcher.rb:
```diff
@@ -106,6 +106,7 @@ def https_app(https_port, plugins = https_plugins)
         :SSLOptions => ssl_options,
         :SSLCiphers => CIPHERS - Proxy::SETTINGS.ssl_disabled_ciphers,
         :daemonize => false,
+        :AccessLog => [],
```
It took me a moment to understand how this was to unify it. Would it make sense to have a `base_app_settings` that returns all common options between HTTP and HTTPS? Things like `Logger`, `ServerSoftware` etc should be there and then merged with HTTP(S) specific options (`Port`, SSL settings).

adamruzicka replied:
I wanted to make the changes as small as possible, but sure, done

ekohl reviewed (commented):
If you want to make it clear, you can introduce a commit that introduces `base_app_settings` without `AccessLog` (as `Refs #35921`) and then one that sets it in base_app_settings.

I also wonder about a clear description for those who don't know what `AccessLog` does. Perhaps it should say something like "Prior to this change http access logs weren't logged while https logs were. After this, neither logs access logs." At least, that's if I properly understood the change. Did I?

adamruzicka committed:
Refs #35921 - Unify use of common app settings

Common options were previously duplicated between http and https
application settings.

adamruzicka committed:
Fixes #35921 - Disable access logging for https endpoint

Previously, access logging was enabled for https endpoint, but not for
the http one. This commit settles things on no access logging for both.

adamruzicka commented:
Ping

ekohl reviewed (approved):
Thanks!
